apps/etterlatte-beregning/src/main/kotlin/beregning/InntektAvkortingService.kt

```diff
@@ -49,28 +49,45 @@ object InntektAvkortingService {
         beregninger: List<Beregningsperiode>,
         avkortingGrunnlag: List<AvkortingGrunnlag>
     ): List<AvkortetYtelse> {
-        val avkortingsgrunnlag = periodisertBruttoYtelseOgAvkorting(beregninger, avkortingGrunnlag)
-        return avkortingsgrunnlag.map { grunnlag ->
+        val avkortedeYtelser = mutableListOf<AvkortetYtelse>()
+
+        val regelgrunnlag = periodisertBruttoYtelseOgAvkorting(beregninger, avkortingGrunnlag)
+        regelgrunnlag.forEachIndexed { grunnlagIndeks, grunnlag ->
             val resultat = avkorteYtelse.eksekver(grunnlag, RegelPeriode(grunnlag.fom.atDay(1)))
             when (resultat) {
                 is RegelkjoeringResultat.Suksess -> {
-                    resultat.periodiserteResultater.map { periodisertResultat ->
-                        AvkortetYtelse(
-                            periode = Periode(
-                                fom = YearMonth.from(periodisertResultat.periode.fraDato),
-                                tom = periodisertResultat.periode.tilDato?.let { YearMonth.from(it) }
-                            ),
-                            ytelseEtterAvkorting = periodisertResultat.resultat.verdi,
-                            tidspunkt = Tidspunkt.now(),
-                            regelResultat = periodisertResultat.toJsonNode()
+                    val periodisertResultatIterator = resultat.periodiserteResultater.iterator()
+                    while (periodisertResultatIterator.hasNext()) {
```

Comment:
Hvorfor iterator?

Reply:
Jeg trodde jeg trengte å vite om man var siste element i innerste loop for å vite at dato til og med måtte hentes fra neste element i ytterste loop.
Men har overtenkt dette. Det jeg skulle gjøre var å sette tom i regelkjøringsperiode.